### PR TITLE
[READY] Multilayer plumbing machinery

### PR DESCRIPTION
**Pull request by Time-Green**
## About The Pull Request

Makes plumbing machinery work on more layers than just the default one.

You can use plungers in-hand to activate LAYER MODE. Hitting plumbing stuff with it changes it to that layer
You can alt-click it to change the layer

## Why It's Good For The Game

Some guy in discord asked. Makes it so factories take up less space since now there's an actual good functional reason to use layers.

## Changelog
:cl:
add: Plungers can now change plumbing machinery layers. Simply use it in-hand once and then alt-click the plunger to select a layer. 
/:cl:

### CODE REVIEW

**Comment by TheBonded** [comment]
Will we get those atmos layer converge point things?

**Comment by MrMelbert** [comment]
image
please tell me you made the layer adapters less janky and broken

**Comment by Winterous** [comment]
Plumbing layers need to be removed and replaced with overlapping coloured ducts. While this is a good idea for the current system, the current system is horribly broken.

**Comment by Time-Green** [comment]
> image
> please tell me you made the layer adapters less janky and broken

They’re not broken, just jank. Duct code causes me immense pain. At least as far as I know, bugs are rarely reported

**Review by Twaticus — APPROVED** [review]
makes me think of a caramel apple haha

File: code/game/objects/structures/lavaland/geyser.dm [hunk 1]
```diff
@@ -79,12 +95,28 @@
...
+
+	layer_mode = !layer_mode
+
+	if(!layer_mode)
+		icon_state = initial(icon_state)
+	else
+		icon_state = layer_mode_sprite
+
+/obj/item/plunger/AltClick(mob/user)
+	var/new_layer = input("Select a layer", "Layer") as null|anything in layers
```
Comment (AnturK) [inline code/game/objects/structures/lavaland/geyser.dm:109]: Ghost/stun etc checks here.

**Review by AnturK — CHANGES_REQUESTED** [review]
Could probably use some feedback on plunger, not sure icon is enough.

**Comment by Time-Green** [comment]
Thanks for the review!

**Comment by Time-Green** [comment]
@AnturK Could you leave your final judgement?

**Pull request merged**